charmcraft/check-libraries/action.yaml

```diff
@@ -0,0 +1,70 @@
+name: "Check if charm libraries are updated"
+description: >
+  Check if - in the current pull request - charm libraries are updated, 
+  and add the relevant label to the PR.
+author: "Canonical"
+inputs:
+  path:
+    description: >
+      The location of the Charmcraft project.
+
+      Defaults to the base of the repository.
+    default: '.'
+  github_token:
+    description: >
+      The GitHub token to use to update the pull requests labels.
+    required: true
+  okay_label:
+    description: >
+      PR label to indicate that charm libraries are updated to their
+      latest LIBPATCH version.
+    default: "Charm Libraries: OK"
+  outdated_label:
+    description: >
+      PR label to indicate thta charm libraries are outdated, and new
+      versions can be fetched.
+    default: "Charm Libraries: Out of sync"
+  charmcraft_channel:
+    description: Install the Charmcraft snap from a specific channel.
+    default: 'latest/stable'
+
+outputs:
+  charms:
+    description: The names of any charm files created.
+    value: ${{ steps.pack-charm.outputs.charms }}
+  charmcraft-revision:
+    description: The revision of Charmcraft used for the pack.
+    value: ${{ steps.setup.outputs.charmcraft-revision }}
+  lxd-revision:
+    description: The revision of lxd used for the pack.
+    value: ${{ steps.setup.outputs.lxd-revision }}
+
+runs:
+  using: 'composite'
+  steps:
+    - id: setup
+      shell: bash
+      env:
+        CHANNEL: ${{ inputs.charmcraft_channel }}
+      run: |
+        sudo snap install charmcraft --classic --channel="$CHANNEL"
+    - id: lib-check
+      shell: bash
+      working-directory: ${{ inputs.path }}
+      env:
+        GH_TOKEN: ${{ secrets.GITHUB_TOKEN }}
+        PR_NUMBER: ${{ github.event.number }}
+        OKAY: "${{ inputs.okay_label }}"
+        OUTDATED: "${{ inputs.outdated_label }}"
+      run: |
+        fetch_lib="$(charmcraft fetch-lib)"
+        # If charm libraries are not up-to-date
+        if echo "$fetch_lib" | grep -qE "not found in Charmhub|updated to version|has local changes"; then
+          gh pr edit "$PR_NUMBER" --remove-label "$OKAY" --add-label "$OUTDATED"
```

Comment:
(No action required, just a question)

Do we have the gh CLI on our self-hosted runners? I know the GH-hosted runners do, but I'm thinking this will be widely used inside Canonical on our self-hosted ones.

Reply:
I asked and yes, it's there as well; only thing is it's installed from `apt`, so the versions might not match (GH runners likely have newer versions)